notificationWorker/upvs/message.go

```diff
@@ -0,0 +1,141 @@
+package upvs
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+
+	"github.com/slovak-egov/einvoice/pkg/random"
+)
+
+type SKTalkMessage struct {
+	XMLName         xml.Name `xml:"http://gov.sk/SKTalkMessage SKTalkMessage"`
+	EnvelopeVersion string
+	Header          Header
+	Body            Body
+}
+
+type Header struct {
+	MessageInfo MessageInfo
+}
+
+type MessageInfo struct {
+	Class            string
+	PospID           string
+	PospVersion      string
+	MessageID        string
+	CorrelationID    string
+	ChannelInfoReply ChannelInfoReply
+}
+
+type ChannelInfoReply struct {
+	Channel Channel
+}
+
+type Channel struct {
+	ChannelInfoURI string
+}
+
+type Body struct {
+	MessageContainer MessageContainer
+}
+
+type MessageContainer struct {
+	XMLName        xml.Name `xml:"http://schemas.gov.sk/core/MessageContainer/1.0 MessageContainer"`
+	MessageId      string
+	SenderId       string
+	RecipientId    string
+	MessageType    string
+	MessageSubject string
+	Objects        []Object
+}
+
+type Object struct {
+	XMLName       xml.Name `xml:"Object"`
+	Id            string   `xml:"Id,attr"`
+	Class         string   `xml:"Class,attr"`
+	MimeType      string   `xml:"MimeType,attr"`
+	Encoding      string   `xml:"Encoding,attr"`
+	Name          string   `xml:"Name,attr"`
+	GeneralAgenda *GeneralAgenda
+	Value         *string `xml:",innerxml"`
+}
+
+type GeneralAgenda struct {
+	XMLName xml.Name `xml:"http://schemas.gov.sk/form/Doc.GeneralAgenda/1.2 GeneralAgenda"`
+	Subject string   `xml:"subject"`
+	Text    string   `xml:"text"`
+}
+
+func generateId() string {
+	return fmt.Sprintf("%s-%s-%s-%s-%s", random.Digits(8), random.Digits(4), random.Digits(4), random.Digits(4), random.Digits(12))
+}
+
+func CreateMsg(senderUri, recipientUri, subject, text, invoiceName string, xmlFile, pdfFile []byte) (string, error) {
```

Comment:
this is not really general `CreateMsg` function so no need to have params `subject`, `text` or `invoiceName` rather create text here and pass just `invoiceId`.